New package: NamingConventions v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: NamingConventions
- Repository: https://github.com/raphasampaio/NamingConventions.jl
- Created by: @raphasampaio
- Version: v0.1.0
- Commit: 97a21e7a6621a66e31d45ac30278618e2c71b467
- Reviewed by: @raphasampaio
- Reference: https://github.com/raphasampaio/NamingConventions.jl/commit/97a21e7a6621a66e31d45ac30278618e2c71b467#commitcomment-146324739
- Description: Create and convert naming conventions
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd62fc2ff9c6477e520d4f4c90e0fba67a257ed220844001f01e444854f363a0242515e19364836c1c5b3ca53f740d245ba2b0f856b6347554f3c2ac2c10914858929629c4a43c348707ff89425eebb563c025bff744c23ae8fed8f50d2b00df32c703184f81e805743d4c8e6d19a80aefc938572b7c52024290233ca664598406bc647573b9f0b516006dd571b77e49e8dabc935fad2dbc82559b95bfbad7140205f5debfecee3eb630f3cb104bffd641354bf165e6538d29bd8468baada83cb64 -->

tecosaur commented:
I will say, just from a glance, I don't like how this package breaks the API of `convert`:

```
  convert(T, x)

  Convert x to a value of type T.
```

Is there a particular reason why you've opted to register this new package instead of going with something like https://github.com/acxz/StringCases.jl?

tecosaur commented:
There has been recent-ish discussion about simply replacing https://github.com/djsegal/StringCases.jl with https://github.com/acxz/StringCases.jl :slightly_smiling_face: 

I'm being a pain here because it seems to me that case conversion is an area where it would be nice to have one solid package, if possible (and I'm a fan of consolidating effort).

It seems that the newer StringCases.jl has all the capabilities we might want, and with a bit of work could perhaps have an API that looks more like this:
```
recase(CamelCase, "...")
recase(ScreamingSnakeCase, "...")
recase(CamelCase => Pascal, "...")
```

Is this direction something I might be able to interest you in?

raphasampaio commented:
[noblock] @tecosaur?

tecosaur commented:
[noblock] Apologies for the delay, I ended up becoming very busy shortly after my last message.

> approach that prioritizes extensibility and simplicity since it is straightforward to create a new case (as shown [here](https://github.com/raphasampaio/NamingConventions.jl?tab=readme-ov-file#example-creating-a-custom-naming-convention)).
> 
> Additionally, using types to define naming conventions makes the package intuitive and expressive, which aligns with my preference for clean and maintainable code.

I'll happily add [noblock] to my previous comment, but from looking at https://github.com/acxz/StringCases.jl, it would seem to me that it's prioritised extensibility and could support a type-based approach with a little work — I think we could have everything you want and some of the things that acxz's StringCases.jl does better in one package.

This is my last little push towards consolidation :)

raphasampaio commented:
Hi @tecosaur, I appreciate your push for consolidation, but there are several issues with the current alternatives:

- Both acxz/StringCases.jl and NamingConventions.jl are not registered.
- The acxz/StringCases.jl package is currently not passing tests and lacks code coverage, which raises concerns about its reliability.
- The interface in acxz/StringCases.jl to define a new case is not as flexible as defining simple encode and decode functions, which is a key feature of NamingConventions.jl.

My goal is to provide a simpler and intuitive approach that emphasizes clean code and ease of extension, which I believe can complement existing tools rather than compete with them.

```julia
(@v1.10) pkg> activate --temp
  Activating new project at `C:\Users\user\AppData\Local\Temp\jl_z4aEGa`

(jl_z4aEGa) pkg> add https://github.com/acxz/StringCases.jl
  Installing known registries into `C:\Users\user\.julia`
       Added `General` registry to C:\Users\user\.julia\registries
     Cloning git-repo `https://github.com/acxz/StringCases.jl`
    Updating git-repo `https://github.com/acxz/StringCases.jl`
    Updating registry at `C:\Users\user\.julia\registries\General.toml`
   Resolving package versions...
    Updating `C:\Users\user\AppData\Local\Temp\jl_z4aEGa\Project.toml`
  [4bbb9ab4] + StringCases v1.0.0-DEV `https://github.com/acxz/StringCases.jl#main`
    Updating `C:\Users\user\AppData\Local\Temp\jl_z4aEGa\Manifest.toml`
  [4bbb9ab4] + StringCases v1.0.0-DEV `https://github.com/acxz/StringCases.jl#main`
Precompiling project...
  ✗ StringCases
  0 dependencies successfully precompiled in 3 seconds
  1 dependency errored.
  For a report of the errors see `julia> err`. To retry use `pkg> precompile`

(jl_z4aEGa) pkg> test StringCases
     Testing StringCases
      Status `C:\Users\user\AppData\Local\Temp\jl_MXHwJh\Project.toml`
  [4bbb9ab4] StringCases v1.0.0-DEV `https://github.com/acxz/StringCases.jl#main`
  [8dfed614] Test
      Status `C:\Users\user\AppData\Local\Temp\jl_MXHwJh\Manifest.toml`
  [4bbb9ab4] StringCases v1.0.0-DEV `https://github.com/acxz/StringCases.jl#main`
  [2a0f44e3] Base64
  [b77e0a4c] InteractiveUtils
  [56ddb016] Logging
  [d6f4376e] Markdown
  [9a3f8284] Random
  [ea8e919c] SHA v0.7.0
  [9e88b42a] Serialization
  [8dfed614] Test
Precompiling project...
  ✗ StringCases
  0 dependencies successfully precompiled in 3 seconds. 1 already precompiled.

ERROR: The following 1 direct dependency failed to precompile:

StringCases [4bbb9ab4-072e-4495-9ad9-68a1bf4994b3]

Failed to precompile StringCases [4bbb9ab4-072e-4495-9ad9-68a1bf4994b3] to "C:\\Users\\user\\.julia\\compiled\\v1.10\\StringCases\\jl_B045.tmp".
ERROR: LoadError: cannot replace module StringCases during compilation
Stacktrace:
 [1] include(mod::Module, _path::String)
   @ Base .\Base.jl:495
...
in expression starting at C:\Users\rsampaio\.julia\packages\StringCases\usc0O\test\runtests.jl:1
ERROR: Package StringCases errored during testing
```

[noblock]

JuliaRegistrator committed:
New package: NamingConventions v0.1.0

UUID: d2ef9726-cd01-476d-83ec-59819e6084ab
Repo: https://github.com/raphasampaio/NamingConventions.jl.git
Tree: b5c17e497c50d64c3e58e3fcd75337bf33eb912b

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

tecosaur commented:
Well, I'm not convinced, but I'm convinced that at this point pushing any further would just be annoying :stuck_out_tongue: 

I think merging should "just happen" now that all my comments end with:

[noblock]
